### PR TITLE
migrate test_linearizer.py to UOp AST, pt. 2

### PR DESCRIPTION
**Pull request by greg-niemeyer**
Completes the migration of test_linearizer.py to UOp AST, started in #6150.

While the diff here is large, I've performed additional manual verification by diffing the test output vs master with DEBUG=4 and confirmed that the generated ASTs are equivalent.

Note: there are small, cosmetic differences due to
- LazyOp using [UnaryOps.NEG](https://github.com/tinygrad/tinygrad/blob/master/extra/ops.py#L69) in master
- op reuse 
image

Also, for the tests that appear to have generated vs handcoded ASTs, I tried to maintain a comparable level of hand tweaking and formatting adjustments where applicable, except for collapsing the ASTs into a single line (I can do this if desired).

### CODE REVIEW

File: test/test_linearizer.py [hunk 1]
```diff
@@ -1522,10 +1688,9 @@ def test_matvec(self):
     assert k.local_dims == 1
     assert k.upcasted == 1
 
-def helper_linearizer_ast(ast:Union[Tuple[LazyOp, ...], LazyOp, UOp], inputs:List[Tensor], *args, **kwargs):
-  if not isinstance(ast, LazyOp) and not isinstance(ast, UOp): ast = LazyOp(MetaOps.KERNEL, ast)
+def helper_linearizer_ast(ast:UOp, inputs:List[Tensor], *args, **kwargs):
+  assert isinstance(ast, UOp), "ast must be UOp"
```
Comment (Qazalin) [inline test/test_linearizer.py:1692]: nice that is this finally in!

**Comment by greg-niemeyer** [comment]
another note: i left these commented lines as TODO -- didn't really know what to do with them: https://github.com/tinygrad/tinygrad/pull/6228/files#diff-44afb98d4eeb30a23927d9d00e184ae2cc32c230a4e817f23812544b2833c967R808-R816

**Comment by Qazalin** [comment]
> another note: i left these commented lines as TODO -- didn't really know what to do with them: https://github.com/tinygrad/tinygrad/pull/6228/files#diff-44afb98d4eeb30a23927d9d00e184ae2cc32c230a4e817f23812544b2833c967R808-R816


@greg-niemeyer can you change the ASTs for it and uncomment? They're green now.

**Comment by greg-niemeyer** [comment]
> @greg-niemeyer can you change the ASTs for it and uncomment? They're green now.

done

**Pull request merged**